// ng2-materialize/dist/select/select-container/select-container.component.js
import { Component, ContentChild } from '@angular/core';
import { NgControl } from '@angular/forms';
import { MzValidationComponent } from '../../validation/validation.component';
import { MzSelectDirective } from '../select.directive';
var MzSelectContainerComponent = (function () {
    function MzSelectContainerComponent() {
    }
    MzSelectContainerComponent.prototype.ngAfterViewInit = function () {
        this.initControlSubscription();
        this.initSelectSubscription();
    };
    MzSelectContainerComponent.prototype.ngOnDestroy = function () {
        this.removeControlSubscription();
        this.removeSelectSubscription();
    };
    MzSelectContainerComponent.prototype.initControlSubscription = function () {
        var _this = this;
        if (this.ngControl) {
            this.mzSelectDirective.disabled = this.ngControl.control.disabled;
            this.statusChangesSubscription = this.ngControl.control.statusChanges.subscribe(function (status) {
                // to handle enabling/disabling formControl
                var disabled = status === 'DISABLED';
                if (disabled !== _this.mzSelectDirective.disabled) {
                    _this.mzSelectDirective.disabled = disabled;
                    _this.mzSelectDirective.handleDisabled();
                }
            });
            this.selectValueSubscription = this.ngControl.valueChanges.subscribe(function (value) {
                var inputValue = _this.mzSelectDirective.inputElement.val();
                var selectedOptionText = _this.mzSelectDirective.selectElement
                    .children('option:selected')
                    .text();
                // synchronize input and select when value changes programmatically
                if (inputValue !== selectedOptionText) {
                    _this.mzSelectDirective.inputElement.val(selectedOptionText);
                    var dropdownOptions = _this.mzSelectDirective.inputElement
                        .siblings('ul.dropdown-content')
                        .children('li');
                    dropdownOptions
                        .removeClass('active selected');
                    dropdownOptions
                        .filter(function (index, element) { return element.textContent === selectedOptionText; })
                        .filter(function (index, element) { return !element.classList.contains('disabled'); })
                        .addClass('active');
                    if (_this.mzValidationComponent) {
                        _this.mzValidationComponent.setValidationState();
                    }
                }
            });
        }
    };
    MzSelectContainerComponent.prototype.initSelectSubscription = function () {
        var _this = this;
        if (this.mzSelectDirective) {
            this.mzSelectDirective.onUpdate.subscribe(function () { return _this.registerOnBlur(); });
            this.registerOnBlur();
        }
    };
    MzSelectContainerComponent.prototype.registerOnBlur = function () {
        var _this = this;
        this.mzSelectDirective.inputElement.on('blur', function () {
            if (_this.ngControl) {
                _this.ngControl.control.markAsTouched();
            }
            if (_this.mzValidationComponent) {
                _this.mzValidationComponent.setValidationState();
            }
        });
    };
    MzSelectContainerComponent.prototype.removeControlSubscription = function () {
        if (this.mzSelectDirective) {
            this.mzSelectDirective.onUpdate.unsubscribe();
            this.mzSelectDirective.inputElement.off();
        }
    };
    MzSelectContainerComponent.prototype.removeSelectSubscription = function () {
        if (this.statusChangesSubscription) {
            this.statusChangesSubscription.unsubscribe();
        }
        if (this.selectValueSubscription) {
            this.selectValueSubscription.unsubscribe();
        }
    };
    return MzSelectContainerComponent;
}());
export { MzSelectContainerComponent };
MzSelectContainerComponent.decorators = [
    { type: Component, args: [{
                selector: 'mz-select-container',
                template: "<div class=\"input-field\"><ng-content></ng-content></div>",
                styles: [".input-field:not(.inline){display:block}"],
            },] },
];
/** @nocollapse */
MzSelectContainerComponent.ctorParameters = function () { return []; };
MzSelectContainerComponent.propDecorators = {
    'mzSelectDirective': [{ type: ContentChild, args: [MzSelectDirective,] },],
    'mzValidationComponent': [{ type: ContentChild, args: [MzValidationComponent,] },],
    'ngControl': [{ type: ContentChild, args: [NgControl,] },],
};
