// ng2-materialize/dist/card/card.component.js
import { ChangeDetectorRef, Component, Directive, ElementRef, Input, } from '@angular/core';
var MzCardComponent = (function () {
    function MzCardComponent(changeDetectorRef, elementRef) {
        this.changeDetectorRef = changeDetectorRef;
        this.elementRef = elementRef;
        this.hasCardAction = true;
        this.hasCardTitle = true;
    }
    MzCardComponent.prototype.ngAfterViewInit = function () {
        this.hasCardTitle = this.hasTitleTagAndNotEmpty();
        this.hasCardAction = this.hasActionTagAndNotEmpty();
        this.changeDetectorRef.detectChanges();
    };
    MzCardComponent.prototype.hasActionTagAndNotEmpty = function () {
        var cardActionElement = this.elementRef.nativeElement.querySelector('mz-card-action');
        return this.isElementDisplayed(cardActionElement);
    };
    MzCardComponent.prototype.hasTitleTagAndNotEmpty = function () {
        var cardTitleElement = this.elementRef.nativeElement.querySelector('mz-card-title');
        return this.isElementDisplayed(cardTitleElement);
    };
    MzCardComponent.prototype.isElementDisplayed = function (element) {
        return element && element.innerHTML.trim() !== '';
    };
    return MzCardComponent;
}());
export { MzCardComponent };
MzCardComponent.decorators = [
    { type: Component, args: [{
                selector: 'mz-card',
                template: "<div class=\"card {{ backgroundClass }}\" [class.hoverable]=\"hoverable\"><div class=\"card-content {{ textClass }}\"><div class=\"card-title\" *ngIf=\"hasCardTitle\"><ng-content select=\"mz-card-title\"></ng-content></div><p><ng-content select=\"mz-card-content\"></ng-content></p></div><div class=\"card-action\" *ngIf=\"hasCardAction\"><ng-content select=\"mz-card-action\"></ng-content></div></div>",
                styles: [""],
            },] },
];
/** @nocollapse */
MzCardComponent.ctorParameters = function () { return [
    { type: ChangeDetectorRef, },
    { type: ElementRef, },
]; };
MzCardComponent.propDecorators = {
    'backgroundClass': [{ type: Input },],
    'hoverable': [{ type: Input },],
    'textClass': [{ type: Input },],
};
// Declare the tags to avoid error: '<mz-card-x>' is not a known element
// https://github.com/angular/angular/issues/11251
// tslint:disable: directive-selector
var MzCardTitleDirective = (function () {
    function MzCardTitleDirective() {
    }
    return MzCardTitleDirective;
}());
export { MzCardTitleDirective };
MzCardTitleDirective.decorators = [
    { type: Directive, args: [{ selector: 'mz-card-title' },] },
];
/** @nocollapse */
MzCardTitleDirective.ctorParameters = function () { return []; };
var MzCardContentDirective = (function () {
    function MzCardContentDirective() {
    }
    return MzCardContentDirective;
}());
export { MzCardContentDirective };
MzCardContentDirective.decorators = [
    { type: Directive, args: [{ selector: 'mz-card-content' },] },
];
/** @nocollapse */
MzCardContentDirective.ctorParameters = function () { return []; };
var MzCardActionDirective = (function () {
    function MzCardActionDirective() {
    }
    return MzCardActionDirective;
}());
export { MzCardActionDirective };
MzCardActionDirective.decorators = [
    { type: Directive, args: [{ selector: 'mz-card-action' },] },
];
/** @nocollapse */
MzCardActionDirective.ctorParameters = function () { return []; };
